### ISSUE TITLE
How to use browser-perf / protractor-perf in a single page application

### ISSUE THREAD

**Issue by dvotion** [axemclion/browser-perf#53]
Hi,
What is the best way to use browser-perf/protractor-perf in a single page application? When running a protractor file in a SPA it would be nice to get the loading time without the protractor / webdriver overhead to execute commands. Maybe create a custom probe/metric which injects a window.performance.now() command?

gr. Dennis

**Comment by axemclion** [comment]
@dvotion I think this heavily depends on the type of SPA that you have. One way I run it is have a single function that does all the interaction, and run a single execute them for them.

Is there a particular scenario that you have in mind >

**Comment by dvotion** [comment]
@axemclion I tested protractor thoroughly by comparing it to video recordings and researched it a lot. What I found out is that Protractor adds a lot of overhead during the test, it varies from 35% to 160% in some cases. I've read somewhere that Protractor monitors the application and waits for it to be stable, this case shows up a lot when testing on a mobile device. Which is logically because a mobile device has a lot less memory & cpu and for functional testing this is wanted to get a stable test without crashes. For performance testing this is not wanted of course.

I also compared Protractor to Webdriver.io and the latter has 50% less overhead (and imho a nicer way of writing specfiles :) But I couldn't get it to run stable for multiple iterations on the same spec case (timeboxed it)

So I finally solved it by decorating a logcollector in the app in the "before" function of Protractor. This decorator places performance.marks so I can calculate the real durationtime at the end. This works great and removes the overhead in my timings caused by Protractor.
